ketcher-autotests/tests/Templates/Custom-Templates/custom-templates.spec.ts

```diff
@@ -0,0 +1,226 @@
+    await page.getByRole('button', { name: 'Aromatics (18)' }).click();
+    await page.getByTitle('Azulene').getByRole('button').click();
+    await takeEditorScreenshot(page);
+    await page.getByRole('button', { name: 'Edit' }).click();
+  });
+  test('Edit templates - Close window', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1700 
+    Description:
+    Make any change(s) in the window. Click the 'Apply' button.
+    */
+    await pressButton(page, STRUCTURE_LIBRARY_BUTTON_NAME);
+    await selectAzuleneOnTemplateLibrary(page);
+    await page.getByTestId('close-icon').click();
+    await selectAzuleneOnTemplateLibrary(page);
+    await page.getByRole('button', { name: 'Cancel' }).click();
+    await selectAzuleneOnTemplateLibrary(page);
+    await page.getByRole('button', { name: 'Edit' }).click();
+  });
+
+  test('Edit templates - Template Name', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1701 
+    Description:
+    Click the 'Custom Template' button.
+    Paste in the 'Molecule name' field text (more than 128 symbols).
+    */
+    await openEditDialogForTemplate(page, TemplateLibrary.Azulene);
+    await page.getByPlaceholder('template').click();
+    await page
+      .getByPlaceholder('template')
+      .fill(
+        'My new template for everyone who want to create new table with more than 128 symbols of elements like Azulene with merged Cyclopentadiene',
+      );
+  });
+
+  test('Edit templates -  Greek symbols in Template Name', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1707 
+    Description: The edited template has the 'γ-template name' (Greek symbol) name in the Template Library.
+    */
+    await openEditDialogForTemplate(page, TemplateLibrary.Azulene);
+    await page.getByPlaceholder('template').click();
+    await page.getByPlaceholder('template').fill('γ-template name');
+  });
+
+  test('Edit templates -  Attachment atom and bond', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1708
+    Description: The info text 'Atom Id: xx; Bond Id: yy' contains the ids of the new attachment atom and bond.
+    */
+    await openEditDialogForTemplate(page, TemplateLibrary.Azulene);
+    await page.getByPlaceholder('template').click();
+    await page.getByRole('dialog').getByTestId('canvas').click();
+  });
+
+  test('Templates Library - Template attachment', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1720
+    Description: The template is attached to the structure by the defined attachment bond.
+    */
+    await openEditDialogForTemplate(page, TemplateLibrary.Azulene);
+    await page.getByPlaceholder('template').click();
+    await page.getByRole('dialog').getByTestId('canvas').click();
+    await takeEditorScreenshot(page);
+    await page.getByRole('button', { name: 'Edit' }).click();
+    await page.getByRole('tab', { name: 'Template Library' }).click();
+    await page.getByRole('button', { name: 'Aromatics (18)' }).click();
+    await page
```

Comment:
Can we use testId or something without  `.filter({ hasText: 'Created with Raphaël 2.3.0' })`?
if Raphael version is updated, then test will fail

Reply:
Done.